### DISCUSSION TITLE
Airflow dag timeout for emr on eks operator (EMRContainerOperator)

### DISCUSSION THREAD

**Discussion by msinha251** [apache/airflow#21367] [General]
### Apache Airflow version

2.2.2 (latest released)

### What happened

We are trying to use airflow EMRContainerOperator for EMR on EKS from airflow DAG.
And following the the dag from [exmaple_emr_on_eks_dag](https://airflow.apache.org/docs/apache-airflow-providers-amazon/2.2.0/_modules/airflow/providers/amazon/aws/example_dags/example_emr_eks_job.html)
DAG is creating the pod on eks cluster but stuck with below airflow ui logs and not proceeding further and after sometime ending up with **timeout**

Error from airflow ui logs:

`[2021-12-10, 00:10:15 UTC] Airflow Connection: aws_conn_id=aws_default`
`[2021-12-10, 00:10:15 UTC] Unable to use Airflow Connection for credentials.`
`[2021-12-10, 00:10:15 UTC] Fallback on boto3 credential strategy`
`[2021-12-10, 00:10:15 UTC] Creating session using boto3 credential strategy region_name=None`

On testing the task with below command from airflow web container, stuck on below and same as above after sometime ending up with timeout:
`airflow tasks test emr_eks_pi_job start_job 2021-01-01`

Error from airflow tasks test command:

`[2021-12-10 00:08:09,139] {base_aws.py:201} INFO - Doing sts_client.assume_role to role_arn=arn:aws:iam::<------------>:role/EmrOnEksRole (role_session_name=airflow_aws_default)`
`[2021-12-10 00:08:09,451] {base_aws.py:152} INFO - New credentials expiry_time:2021-12-10T01:08:09+00:00`

### What you expected to happen

DAG should run and complete successfully.

### How to reproduce

Deploy airflow 2.2.2
Copy the DAG from [exmaple_emr_on_eks_dag](https://airflow.apache.org/docs/apache-airflow-providers-amazon/2.2.0/_modules/airflow/providers/amazon/aws/example_dags/example_emr_eks_job.html)
Run the dag or run _airflow tasks test command_

### Operating System

Debian 10

### Versions of Apache Airflow Providers

Airflow: 2.2.2
apache-airflow-providers-amazon==2.2.0 
also tried with 
apache-airflow-providers-amazon==2.3.0

### Deployment

Other

### Deployment details

Using EKS and argocd for deplyoment.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by msinha251** [comment]
Is anything is pending on my above issue ?

**Comment by potiuk** [comment]
I think not. It just waits for someone willing to take a look and fix - seems like it neeeds someeone to take a look and try to resolve it - maybe you would like to take a stab on fixing it? 

This is how it works here - if there are people who would like to work on it and fix it - then good, but if not, the best way to approach it is to try to fix it yourself. Seems like a specific AWS issue, so there is a chance someone who wrote this part will take a look soner or later (but there is no guarantee). Good first issue means that it sounds like a simple one for someone who knows AWS and uses it (like you).

**Comment by burakovsky** [comment]
I had a similar problem and the issue was with Airflow IAM permissions. According to the [official documentation](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/iam-execution-role.html), your Airflow user (or role) must have `emr-containers:StartJobRun` permissions to run EMR job in EKS. But for Airflow, it's also required to have `emr-containers:DescribeJobRun` (and optional `emr-containers:CancelJobRun` for job canceling). The final IAM policy which works for me looks like this:
```
{
    "Version": "2012-10-17"
    "Statement": [
        {
            "Sid": ""
            "Effect": "Allow",
            "Action": [
                "emr-containers:StartJobRun"
            ],
            "Resource": "arn:aws:emr-containers:REGION:AWS_ACCOUNT_ID:/virtualclusters/VIRTUAL_CLUSTER_ID",
            "Condition": {
                "StringEquals": {
                    "emr-containers:ExecutionRoleArn": "EXECUTION_ROLE_ARN"
                }
            }
        },
        {
            "Sid": ""
            "Effect": "Allow",
            "Action": [
                "emr-containers:DescribeJobRun",
                "emr-containers:CancelJobRun"
            ],
            "Resource": "arn:aws:emr-containers:REGION:AWS_ACCOUNT_ID:/virtualclusters/VIRTUAL_CLUSTER_ID/*",
        }
    ]
}
``` 

Also, if you send EMR job logs to CloudWatch as is configured in the [example](https://airflow.apache.org/docs/apache-airflow-providers-amazon/2.2.0/_modules/airflow/providers/amazon/aws/example_dags/example_emr_eks_job.html), be sure, that CloudWatch log group is created or your execution role has permissions to create new CloudWatch log group.

I tested it on Airflow 2.1.4, probably for the latest version it works a bit differently.

**Comment by msinha251** [comment]
Thanks @burakovsky my aws user permission was missing "emr-containers:StartJobRun" and "emr-containers:CancelJobRun" permissions.

Post added above two permissions, it worked like charm :)

**Comment by potiuk** [comment]
Converted it into discussion as it seems it's been a configuration issue.
